src/processor/graph-notifier.service.ts

```diff
@@ -0,0 +1,136 @@
+import { InjectRedis } from '@liaoliaots/nestjs-redis';
+import { InjectQueue, Processor, WorkerHost } from '@nestjs/bullmq';
+import { Injectable, Logger } from '@nestjs/common';
+import { Job, Queue } from 'bullmq';
+import Redis from 'ioredis';
+import { MILLISECONDS_PER_SECOND } from 'time-constants';
+import { RegistryError } from '@polkadot/types/types';
+import { BlockchainService } from '#app/blockchain/blockchain.service';
+import { BlockchainConstants } from '#app/blockchain/blockchain-constants';
+import { ITxMonitorJob } from '#app/interfaces/monitor.job.interface';
+import { IGraphUpdateJob } from '#app/interfaces/graph-update-job.interface';
+
+@Injectable()
+@Processor('graphTxMonitorQueue')
+export class GraphNotifierService extends WorkerHost {
+  private logger: Logger;
+
+  constructor(
+    @InjectRedis() private cacheManager: Redis,
+    @InjectQueue('graphUpdateQueue') private reconnectionQueue: Queue,
+    private blockchainService: BlockchainService,
+  ) {
+    super();
+    this.logger = new Logger(this.constructor.name);
+  }
+
+  async process(job: Job<ITxMonitorJob, any, string>): Promise<any> {
+    this.logger.log(`Processing job ${job.id} of type ${job.name}`);
+    try {
+      const numberBlocksToParse = BlockchainConstants.NUMBER_BLOCKS_TO_CRAWL;
+      const txCapacityEpoch = job.data.epoch;
+      const previousKnownBlockNumber = (await this.blockchainService.getBlock(job.data.lastFinalizedBlockHash)).block.header.number.toBigInt();
+      const currentFinalizedBlockNumber = await this.blockchainService.getLatestFinalizedBlockNumber();
+      const blockList = Array.from(
+        { length: Math.min(Number(numberBlocksToParse), Number(currentFinalizedBlockNumber) - Number(previousKnownBlockNumber)) },
+        (_, index) => previousKnownBlockNumber + BigInt(index) + 1n,
+      );
+      const txResult = await this.blockchainService.crawlBlockListForTx(job.data.txHash, blockList, [{ pallet: 'system', event: 'ExtrinsicSuccess' }]);
+      if (!txResult.found) {
+        this.logger.error(`Tx ${job.data.txHash} not found in block list`);
+        throw new Error(`Tx ${job.data.txHash} not found in block list`);
+      } else {
+        // Set current epoch capacity
+        await this.setEpochCapacity(txCapacityEpoch, BigInt(txResult.capacityWithDrawn ?? 0n));
```

Comment:
Re-linking comment thread about the Capacity epoch:

https://github.com/AmplicaLabs/reconnection-service/pull/97/files/aef137675a4821b3d75e54baca5b2bfcf2f8b7ff#r1551730679